### DISCUSSION TITLE
triqs mpi environment

### DISCUSSION THREAD

**Discussion by daisychoco** [TRIQS/dft_tools#224] [General]
Hi,
I'm learning dft_tools through web tutorial page.
When I ran "import triqs.utility.mpi as mpi"  I got

"Warning: could not identify MPI environment!
Starting serial run at: 2022-10-13 16:13:27.174887"

I am wondering how i can set MPI environment.
If I type "mpirun", I got 
"mpirun could not find anything to do.
It is possible that you forgot to specify how many processes to run
via the "-np" argument." and I assumed mpi environment is already set.

Could you help to set triqs mpi enviorment ?

Regards,

**Comment by the-hampel** [comment]
Hi @daisychoco ,
Since this is not strictly and issue with the code I moved this to the discussions section. 

The message itself is not a problem for running anything, it is only a warning. Can you explain a bit how you are running your python code? Did you create a python script? Are you running inside of jupyter? 

In general, to run a triqs python script via MPI you have to create first a python script that uses the import:
```
import triqs.utility.mpi as mpi
```
Next, you should execute the script like this to run via MPI:
```
mpirun -np 4 python your_triqs_script.py
```
Where you can change `4` with whatever number of mpi ranks you would like to allocate. Can you try this and show any error messages you encounter? It is very unlikely that your MPI environment is not yet setup correctly, since TRIQS should automatically check this upon installation.

Best,
Alex

**Comment by daisychoco** [comment]
Thank you, Alex, for your suggestion that works.
I missed the "mpirun -np 4" part. Basically, I  executed a single job but
worried about parallel running..
What a D.....  Thank you for your kind reply.

Best,
Yongbin
